c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -790,6 +791,10 @@ cudaError_t cudaMallocMaybeCapturing(void** p, size_t size) {
 
 static std::string reportProcessMemoryInfo(int device) {
 #ifdef PYTORCH_C10_DRIVER_API_SUPPORTED
+  void* handle_1 = dlopen("libnvidia-ml.so.1", RTLD_LAZY);
```

Comment:
this would be calling `dlopen` on each invocation?

Reply:
Need a static once flag.

Reply:
That's a good catch, could we make this static @Fuzzkatt ?

Reply:
Good idea, make a fix for this